pkgs/tools/X11/xidlehook/default.nix

```diff
@@ -0,0 +1,30 @@
+{ lib, rustPlatform, fetchFromGitHub, x11, xorg, libpulseaudio, pkgconfig, makeWrapper }:
+
+rustPlatform.buildRustPackage rec {
+  name = "xidlehook-${version}";
+  version = "0.4.5";
+
+  src = fetchFromGitHub {
+    owner = "jD91mZM2";
+    repo = "xidlehook";
+    rev = version;
+
+    sha256 = "1jdy85sh7w8drcahx33swqwnv2ik7rpxl0hc07waxw3iwwk63m66";
+  };
+
+  cargoSha256 = "0a1bl6fnfw6xy71q3b5zij52p9skylj1ivqj8my44bfsid2qfn7d";
+
+  buildInputs = [ x11 xorg.libXScrnSaver libpulseaudio ];
+  propagatedBuildInputs = [ pkgconfig makeWrapper ];
```

Comment:
Could you move these to `nativeBuildInputs`? :)

Reply:
What's the difference?

Reply:
`nativeBuildInputs` are the ones used on the build host and thus have to be compiled for it's architecture. So when a x86 host cross-compiles something for aarch64, the build tools have to be built for x86 even though the target is aarch64.

Reply:
Here is some more information in the  [manual](https://nixos.org/nixpkgs/manual/#ssec-stdenv-dependencies) :)

Reply:
Awesome, thanks!